src/main/java/seedu/address/commons/util/StringUtil.java

```diff
@@ -35,6 +38,37 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
                 .anyMatch(preppedWord::equalsIgnoreCase);
     }
 
+    /**
+     * Counts how many times the substring appears in the larger string.
+     * Note that the code only counts non-overlapping matches.
+     * A {@code null} or empty ("") String input returns {@code 0}.
+     * <br>examples:<pre>
+     * StringUtil.countMatches(null, *)       = 0
+     * StringUtil.countMatches("", *)         = 0
+     * StringUtil.countMatches("abba", null)  = 0
+     * StringUtil.countMatches("abba", "")    = 0
+     * StringUtil.countMatches("abba", "a")   = 2
+     * StringUtil.countMatches("abba", "ab")  = 1
+     * StringUtil.countMatches("abba", "xxx") = 0
+     * StringUtil.countMatches("ababa", "aba") = 1
+     * </pre>
+     * @param str  the String to check, may be null
+     * @param sub  the substring to count, may be null
+     * @return the number of occurrences, 0 if either String is {@code null}
+     */
+    public static int countMatches(String str, String sub) {
+        if (str.isEmpty() || sub.isEmpty()) {
+            return 0;
+        }
+        int count = 0;
+        int idx = 0;
+        while ((idx = str.indexOf(sub, idx)) != -1) {
```

Comment:
I like that this is so succinct, but others who are less proficient/new to Java might find this hard to digest, since it condenses a few steps into one. Do you want to consider writingn this on separate lines?